src/googleCloud/DatasetSelector.js

```diff
@@ -0,0 +1,169 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import DicomStorePicker from './DicomStorePicker';
+import DatasetPicker from './DatasetPicker';
+import ProjectPicker from './ProjectPicker';
+import LocationPicker from './LocationPicker';
+import './googleCloud.css';
+
+export default class DatasetSelector extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      project: null,
+      location: null,
+      dataset: null,
+      unloading: false,
+    };
+
+    this.onProjectSelect = this.onProjectSelect.bind(this);
+    this.onLocationSelect = this.onLocationSelect.bind(this);
+    this.onDatasetSelect = this.onDatasetSelect.bind(this);
+    this.onDicomStoreSelect = this.onDicomStoreSelect.bind(this);
+    this.onProjectClick = this.onProjectClick.bind(this);
+    this.onLocationClick = this.onLocationClick.bind(this);
+    this.onDatasetClick = this.onDatasetClick.bind(this);
+  }
+
+  static propTypes = {
+    id: PropTypes.string,
+    event: PropTypes.string,
+    oidcKey: PropTypes.string,
+    canClose: PropTypes.string,
+    setServers: PropTypes.func.isRequired,
+  };
+  static defaultProps = {};
+
+  onProjectSelect(project) {
+    this.setState({
+      project: project,
+    });
+  }
+
+  onLocationSelect(location) {
+    this.setState({
+      location: location,
+    });
+  }
+
+  onDatasetSelect(dataset) {
+    this.setState({
+      dataset: dataset,
+    });
+  }
+
+  onProjectClick() {
+    this.setState({
+      dataset: null,
+    });
+    this.setState({
+      location: null,
+    });
+    this.setState({
+      project: null,
+    });
+  }
+
+  onLocationClick() {
+    this.setState({
+      dataset: null,
+    });
+    this.setState({
+      location: null,
+    });
+  }
+
+  onDatasetClick() {
+    this.setState({
+      dataset: null,
+    });
+  }
+
+  onDicomStoreSelect(dicomStoreJson) {
+    const dicomStore = dicomStoreJson.name;
+    const parts = dicomStore.split('/');
+    const result = {
+      wadoUriRoot: `https://healthcare.googleapis.com/v1beta1/${dicomStore}/dicomWeb`,
```

Comment:
Could we extract the `https://healthcare.googleapis.com/v1beta1` part of this into a variable somewhere so we only have to update it in one place once it gets to beta2, etc?

Reply:
done